cpp/src/delayed_submission.cpp

```diff
@@ -35,26 +35,25 @@ void DelayedSubmissionCollection::process()
       toProcess = std::move(_collection);
```

Comment:
What does `std::move` do to `_collection`? Does it just drain the vector, leaving it "as-if" it were empty again? I guess so.